foreshadow/foreshadow.py

```diff
@@ -555,3 +554,32 @@ def configure_multiprocessing(self, n_job: int = 1) -> NoReturn:
 
         """
         self.X_preparer.cache_manager["config"][ConfigKey.N_JOBS] = n_job
+
+    # TODO I'm not sure if this is a good idea to allow the user to tune the
+    #  sampling criteria as it's more internal comparing to the intent and
+    #  multiprocessing options. We can discuss more on this.
+    def configure_sampling(
+        self,
+        enable_sampling=True,
+        sampling_fraction: float = 0.2,
+        replace: bool = False,
+    ) -> NoReturn:  # noqa: S001
+        """Configure the sampling criteria.
+
+        Args:
+            enable_sampling: whether to enable sampling in data cleaning and intent resolving # noqa: E501
+            sampling_fraction: whether to use replacement during sampling
+            replace: the sampling fraction
+
+        Returns:
+
+        """
+        self.X_preparer.cache_manager["config"][
+            ConfigKey.ENABLE_SAMPLING
+        ] = enable_sampling
+        self.X_preparer.cache_manager["config"][
+            ConfigKey.SAMPLING_FRACTION
+        ] = sampling_fraction
+        self.X_preparer.cache_manager["config"][
```

Comment:
Better to make 'config' a constant as well.

Reply:
Done.